tests/test_labels_present.py

```diff
@@ -0,0 +1,68 @@
+from selenium.webdriver.remote.webdriver import WebDriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver import ActionChains
+from selenium.webdriver.common.keys import Keys
+
+web = "https://machinist.flapflap.io/"
+
+
+def test_transition_label_presence(driver: WebDriver):
+    driver.get(web)
+
+    canvas_element = driver.find_element(By.XPATH, '//*[@id="__next"]/main/div[1]')
+    action = ActionChains(driver)
+
+    # Adding q0 theoretically even if bug is present
+    action.move_to_element_with_offset(
+        canvas_element, 100, 100
+    ).double_click().perform()
+
+    # Adding q1 theoretically with bug present
+    action.move_to_element_with_offset(
+        canvas_element, 200, 200
+    ).double_click().perform()
+
+    driver.find_element(By.XPATH, '//*[@id="__next"]/main/div[1]/div/div[2]').click()
+
+    # The transition ball
+    source = driver.find_element(
+        By.XPATH, '//*[@id="__next"]/main/div[1]/div/div[2]/div[2]/div'
+    )
+
+    # Drag to q1
+    target = driver.find_element(By.XPATH, '//*[@id="__next"]/main/div[1]/div/div[3]')
```

Comment:
These kinds of XPATHs are getting pretty gnarly...

Ideally, we can avoid doing complex xpaths. I understand that it is easy though to simply copy/paste the xpath from the browser debugger.

Specifically, the xpaths where we are selecting a specific numbered child (the part where it says `div[3]`). These are very fragile because if for some reason the order of the elements is changed in the UI, then the test breaks.

Anyways, this ends up being technical debt because it makes it more difficult to change the UI without breaking tests.

I think we can merge it, but this should be noted and added to list of technical debt, and in future tests we should try to avoid making fragile selenium selectors

Reply:
It is better to select the element that you are looking for based on a specific and uniquely identifying property of the element, like the text it contains or a class name.

Reply:
Prefer `By.CSS_SELECTOR` to `By.XPATH` when possible too

Reply:
I tried the CSS selector, but they end up really long and confusing just like the XPATH. Sometimes, with the css selector it can't be identified either. I will take your comment into consideration and make the changes later :)